### ISSUE TITLE
heplify on AVAYA

### ISSUE THREAD

**Issue by sebky94** [sipcapture/heplify#131]
Hello, 

I installed heplify on AVAYA SBC for homer sipcapture (server address : 172.19.47.206) , when i execute the heplify i get this message :
command :  ./heplify -hs 172.19.47.206:9060
result : bash: ./heplify: Permission denied

Please explain this to me.

Many thanks.
BR.

**Comment by lmangani** [comment]
If you have rights to do so, `chmod +x ./heplify`

**Comment by sebky94** [comment]
Already done, still the same problem.

**Comment by negbie** [comment]
Normally you need to be root for this kind of operation. Keep in mind that when you run heplify without -i flag it will listen on any interface.

**Comment by sebky94** [comment]
Yes i did it as root.

**Comment by negbie** [comment]
Hmm ok let me reopen this and see if we can fix it. What os and kernel is this AVAYA SBC using?

**Comment by sebky94** [comment]
Thank you. 

Red Hat Enterprise Linux Server release 7.3 (Maipo)

**Comment by negbie** [comment]
Could you paste the output of:
ls -al heplify
uname -a
ip a | grep "<"

**Comment by sebky94** [comment]
[root@DOUSBC etc]# ls -al heplify
ls: cannot access heplify: No such file or directory (because i uploaded the file using SFTP)
[root@DOUSBC etc]# uname -a
Linux DOUSBC 3.10.0-693.46.1.el7.AV1.x86_64 #1 SMP Tue Apr 2 18:51:06 +0530 2019 x86_64 x86_64 x86_64 GNU/Linux
[root@DOUSBC etc]# ip a | grep "<"
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1
2: B2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
3: B1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
4: A2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
5: A1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
6: M2: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
7: M1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
8: A2.12@A2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
9: B2.342@B2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000

this is how i installed heplify : 

I took the install file from here : https://github.com/sipcapture/heplify/releases
and then uploaded it to the SBC using SFTP, then "chmod +x heplify" as root and tried the commands that didn't work.

**Comment by negbie** [comment]
Could you explain this in more detail why you can't use ls? When you can use sftp can't you just ssh into this box?
ls: cannot access heplify: No such file or directory (because i uploaded the file using SFTP)

**Comment by negbie** [comment]
Under which location did you uploaded it? You must be able to do a ls -al heplify

**Comment by negbie** [comment]
From you output I see you are under /etc
Did you uploaded heplify under that path? A better location would be /opt or your home dir.

**Comment by sebky94** [comment]
I uploaded it in ipcs (default user of AVAYA SBC) i wasn't in the right directory as you noticed, now this is the correct result: 
[ipcs@DOUSBC ~]$ ls -al heplify
-rwxr--r-- 1 ipcs ipcs 9812840 Oct 21 11:17 heplify

**Comment by negbie** [comment]
Now do chmod +x heplify
This time in the right directory and watch ls -al heplify

**Comment by sebky94** [comment]
[root@DOUSBC ipcs]# chmod +x heplify
[root@DOUSBC ipcs]# ls -al heplify
-rwxr--r-- 1 ipcs ipcs 9812840 Oct 21 11:17 heplify


Still the same rights?

**Comment by negbie** [comment]
Hmm I would have expected this:
-rwxrwxr-x

1st bit: tells type of file ( - is for regular file )
Next 3 bits: owner permission ( rwx : read, write and execute )
Next 3 bits: group permission ( rwx : read , write and execute )
Next 3 bits: other permission (r-x : read and execute permission )

**Comment by sebky94** [comment]
Yes i expected the rights to change, apparently they didn't. 

Do you think it's because of the redhat release?

**Comment by sebky94** [comment]
I did this, you think it might help? :

[root@DOUSBC ipcs]# chmod 777 heplify
[root@DOUSBC ipcs]# ls -al heplify
-rwxrwxrwx 1 ipcs ipcs 9812840 Oct 21 11:17 heplify

**Comment by sebky94** [comment]
Still the same result : 

[root@DOUSBC ipcs]# ./heplify -hs 172.19.47.206:9060
bash: ./heplify: Permission denied

**Comment by negbie** [comment]
Can you try ./heplify -i A1 or some other interface?

**Comment by sebky94** [comment]
[root@DOUSBC ipcs]# ./heplify -i A1
bash: ./heplify: Permission denied


Do you think it's because it's in he directory ipcs? should i move it to opt?

**Comment by sebky94** [comment]
I tried it and it didn't give an error, and didn't give any output either: 
[root@DOUSBC opt]# ./heplify -i any -hs 172.19.47.206:9060


do i understand from this that it worked? my homer7 server now can receive data?

**Comment by negbie** [comment]
Basicly yes. You can try this to debug:

./heplify -i any -hs "" -e

This will print captured packet only on screen but won't send them any where.

**Comment by negbie** [comment]
I think you should be good to go now. Just check heplify and heplify-server logs. Make you familiar with the debug flags.

**Comment by sebky94** [comment]
Yes i got on screen my SIP data, i see some OPTIONs and SDP data. 

Thank you very much, you can close the issue.

**Comment by negbie** [comment]
Your welcome :)
